### PR TITLE
[Messenger] Wire the Doctrine transport factory when component is enabled

### PR DESCRIPTION
**Pull request by vincenttouzet**
This is related to this PR : https://github.com/symfony/symfony/pull/29007

### CODE REVIEW

File: Resources/config/messenger.xml [hunk 1]
```diff
@@ -11,5 +11,12 @@
         <service id="messenger.middleware.doctrine_transaction" class="Symfony\Bridge\Doctrine\Messenger\DoctrineTransactionMiddleware" abstract="true" public="false">
             <argument type="service" id="doctrine" />
         </service>
+
+        <service id="messenger.transport.doctrine.factory" class="Symfony\Component\Messenger\Transport\Doctrine\DoctrineTransportFactory">
+            <argument type="service" id="doctrine" />
+            <argument type="service" id="messenger.transport.serializer" on-invalid="ignore" />
```
Comment (sroze) [inline Resources/config/messenger.xml]: This needs to be removed now.

**Comment by greg0ire** [comment]
Why target master? Is there a BC-break?

File: Resources/config/messenger.xml (hunk 1 above)
Comment (greg0ire) [reply Resources/config/messenger.xml]: Indeed, there is no second argument: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineTransportFactory.php#L29

**Comment by sroze** [comment]
The class is only available in Symfony 4.3.

**Comment by vincenttouzet** [comment]
@greg0ire No just a new feature for the next version in Symfony 4.3. But maybe I target the wrong branch ? Which one must be the target ?

**Comment by greg0ire** [comment]
Ah maybe the doctrine bundle is special and new features should go on master. Note that branch 1.10.x does not seem to have an incompatibility with symfony 4.3, but I'm not really sure what the workflow is for this bundle, so disregard my comment for now.

**Comment by alcaeus** [comment]
`master` is good here, but I'll take a closer look over the next few days 👍

**Comment by weaverryan** [comment]
Here is the class it's wiring: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineTransportFactory.php

If the user doesn't have symfony/messenger 4.3, then they won't have this class. Normally, I think that's not a problem - because as long as they don't try to use the feature, it won't case an error. But because this is tagged as a transport, wouldn't a Symfony 4.2 user get an error about a missing class if they tried to use any messenger transport? @vincenttouzet can you check that?

**Comment by sroze** [comment]
Indeed, because it's tagged as a transport factory, it will be used... therefore this would fail for a Symfony 4.2 user. We need to add, in a compiler pass, that if the class does not exist, we remove this definition.

**Comment by alcaeus** [comment]
Superseded by #944. Thanks @vincenttouzet!

**Pull request closed without merging**